packages/webapp/src/webview/state/reducers.ts

```diff
@@ -112,6 +115,8 @@ function cloneState(state: AppState): AppState {
  */
 function updateGuidedAnswerTreesReducer(newState: AppState, action: UpdateGuidedAnswerTrees): AppState {
     newState.guidedAnswerTreeSearchResult = action.payload;
+    newState.updatedGuidedAnswerTrees = [...newState.updatedGuidedAnswerTrees, ...action.payload.trees];
```

Comment:
Can't we add the new trees to `newState.guidedAnswerTreeSearchResult.trees`?
I mean something like this
```
const trees = newState.guidedAnswerTreeSearchResult.trees;
newState.guidedAnswerTreeSearchResult = action.payload;
newState.guidedAnswerTreeSearchResult.trees.unshift(...trees);
```

This way we would avoid keeping parts of the trees duplicated and having an additional array on the state.

Reply:
Thanks @Klaus-Keller,  what you suggested is a lot better. I made the changes. I had this issue https://github.com/ankeetmaini/react-infinite-scroll-component/issues/274 and the only way I could make it work was at that time was with a duplicate tree.